File: web/types/wine.ts
export interface ParsedLabel {
  winery: string | null;
  wine_name: string | null;
  vintage: number | null;
  grape_variety: string | null;
  appellation: string | null;
  alcohol_pct: number | null;
  estate_bottled: boolean | null;
  reserve: boolean | null;
  raw_label_text: string | null;
}

export interface RegionProfile {
  name: string;
  parent_ava: string | null;
  climate_type: string;
  soil_types: string[];
  key_grapes: string[];
  climate_summary: string;
  known_for: string;
}

export interface VintageNote {
  year: number;
  region: string;
  rating: string;
  weather_summary: string;
  aging_potential: string;
}

export interface GrapeProfile {
  variety: string;
  skin_color: string;
  california_character: string;
  flavor_profile: string[];
  structure: string;
  food_pairings: string[];
  peak_drinking_window: string;
}

export interface ProducerProfile {
  name: string;
  founded: number | null;
  region: string;
  philosophy: string;
  tier: string;
  known_wines: string[];
}

export interface WineAnalysis {
  parsed_label: ParsedLabel;
  region_profile: RegionProfile | null;
  producer_profile: ProducerProfile | null;
  vintage_note: VintageNote | null;
  grape_profile: GrapeProfile | null;
  terroir_narrative: string;
  buying_guidance: string;
  geek_notes: string;
  confidence: string;
}

File: api/models/wine.py
from pydantic import BaseModel
from typing import Optional


class LabelParseRequest(BaseModel):
    image_base64: Optional[str] = None
    image_url: Optional[str] = None
    raw_text: Optional[str] = None  # fallback: user types the label manually


class ParsedLabel(BaseModel):
    winery: Optional[str] = None
    wine_name: Optional[str] = None
    vintage: Optional[int] = None
    grape_variety: Optional[str] = None          # e.g. "Cabernet Sauvignon"
    appellation: Optional[str] = None            # e.g. "Napa Valley", "Oakville"
    alcohol_pct: Optional[float] = None
    estate_bottled: Optional[bool] = None
    reserve: Optional[bool] = None
    raw_label_text: Optional[str] = None         # full extracted text


class RegionProfile(BaseModel):
    name: str
    parent_ava: Optional[str]
    climate_type: str                            # e.g. "maritime", "continental"
    soil_types: list[str]
    key_grapes: list[str]
    climate_summary: str
    known_for: str


class ProducerProfile(BaseModel):
    name: str
    founded: Optional[int]
    region: str
    philosophy: str                              # winemaking style/philosophy
    tier: str                                    # "boutique", "mid-size", "large estate"
    known_wines: list[str]


class VintageNote(BaseModel):
    year: int
    region: str
    rating: str                                  # "exceptional", "good", "average", "challenging"
    weather_summary: str
    aging_potential: str


class GrapeProfile(BaseModel):
    variety: str
    skin_color: str                              # "red", "white"
    california_character: str                    # how it expresses in CA specifically
    flavor_profile: list[str]
    structure: str                               # tannin, acid, body description
    food_pairings: list[str]
    peak_drinking_window: str                    # e.g. "5-15 years from vintage"


class WineAnalysis(BaseModel):
    parsed_label: ParsedLabel
    region_profile: Optional[RegionProfile]
    producer_profile: Optional[ProducerProfile]
    vintage_note: Optional[VintageNote]
    grape_profile: Optional[GrapeProfile]
    terroir_narrative: str      # the synthesized story: how all 4 factors shape THIS bottle
    buying_guidance: str        # drink now vs cellar, value assessment
    geek_notes: str             # deep cuts: clonal selection, oak regime, elevation, etc.
    confidence: str             # "high" / "medium" / "low" based on label clarity
